packages/wallets/src/features/cfd/screens/MT5TradeScreen/MT5TradeLink/mobileOsDetect.ts

```diff
@@ -0,0 +1,182 @@
+import UAParser from 'ua-parser-js';
+
+declare global {
+    interface Window {
+        MSStream?: {
+            msClose: () => void;
+            msDetachStream: () => void;
+            readonly type: string;
+        };
+        opera?: string;
+    }
+    interface Navigator {
+        userAgentData?: NavigatorUAData;
+    }
+}
+
+type NavigatorUAData = {
+    brands: { brand: string; version: string }[];
+    getHighEntropyValues(hints: string[]): Promise<HighEntropyValues>;
+    mobile: boolean;
+};
+
+type HighEntropyValues = {
+    model?: string;
+    platform?: string;
+    platformVersion?: string;
+    uaFullVersion?: string;
+};
+
+// Simple regular expression to match potential Huawei device codes
+const huaweiDevicesRegex = /\b([A-Z]{3}-)\b/gi;
+
+// Set of valid Huawei device codes
+const validCodes = new Set([
```

Comment:
Best to move this to deriv-com/utils and reference it from there.

This code already exists in 2 places.